common/tests/filter_test.go

```diff
@@ -0,0 +1,65 @@
+package tests
+
+import (
+	"reflect"
+	"testing"
+
+	"github.com/HackIllinois/api/common/database"
+)
+
+type TestStruct struct {
+	Value1 string `json:"value1"`
+	Value2 string `json:"value2"`
+}
+
+func TestFilterBasic(t *testing.T) {
+
+	params := map[string][]string{
+		"value1": {"test"},
+		"value2": {"foo,bar"},
+	}
+
+	expected_query := map[string]interface{}{
+		"value1": database.QuerySelector{"$in": []string{"test"}},
+		"value2": database.QuerySelector{"$in": []string{"foo", "bar"}},
+	}
+
+	query, err := database.CreateFilterQuery(params, TestStruct{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !reflect.DeepEqual(query, expected_query) {
+		t.Errorf("Wrong order.\nExpected %v\ngot %v\n", expected_query, query)
+	}
+}
+
+type TestStruct2 struct {
+	Hack     string `json:"hackValue"`
+	Illinois int64  `json:"illinoisValue"`
+	Test     bool   `json:"testValue"`
+}
+
+func TestFilterCasting(t *testing.T) {
+
+	params := map[string][]string{
+		"hackValue":     {"foo,bar"},
+		"illinoisValue": {"55,63"},
+		"testValue":     {"true,false,true"},
+	}
+
+	expected_query := map[string]interface{}{
+		"hackvalue":     database.QuerySelector{"$in": []string{"foo", "bar"}},
+		"illinoisvalue": database.QuerySelector{"$in": []int64{55, 63}},
+		"testvalue":     database.QuerySelector{"$in": []bool{true, false, true}},
+	}
+
+	query, err := database.CreateFilterQuery(params, TestStruct2{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !reflect.DeepEqual(query, expected_query) {
+		t.Errorf("Wrong order.\nExpected %v\ngot %v\n", expected_query, query)
+	}
+}
```

Comment:
How about a test for non-existent fields, and duplicate fields?

Reply:
I added a non-existent field test, but maps don't support duplicate keys so I couldn't add a duplicate entry

Reply:
Yup this looks good.